site/en/blog/angular-image-directive/index.md

```diff
@@ -0,0 +1,215 @@
+
+
+    ``` html
+        // in module providers:
+        provideImgixLoader('https://mysite.net/assets/')
+
+        // in markup
+        <img rawSrc="image.png" >
+        <img rawSrc="image2.png" >
+    ```
+
+
+
+    This is equivalent to including the following image tags and reduces the markup developers must include for every image.
+
+
+    ``` html
+        <img src="https://mysite.net/assets/image.png">
+        <img src="https://mysite.net/assets/image2.png">
+    ```
+
+    The image directive provides built-in loaders with optimal configuration for the most popular image CDNs. These loaders will format image URLs automatically to ensure that the recommended image format and compression settings are used for each CDN.  
+
+4. **Built-in errors and warnings**
+
+    In addition to the above built-in optimizations, the directive also has built-in checks to ensure that developers have followed the recommended best practices in the image markup. The image directive performs the following checks.
+
+    1. **Unsized images:** The image directive throws an error if the image markup does not have defined an explicit width and height. Unsized images can cause [layout shifts](https://web.dev/optimize-cls/#images-without-dimensions), affecting the page's Cumulative Layout Shift ([CLS](https://web.dev/cls/)) metric. The recommended best practice to prevent this is that images should have `width` and `height` attributes specified. 
+    
+    2. **Aspect ratio:** The image directive throws an error to let developers know if the aspect ratio of the `width`:`height` defined in the HTML is not close to the actual aspect ratio of the rendered image. This can cause the image to look distorted on screen. This can happen if 
+        1. You have defined the wrong dimensions (width or height) by mistake or
+        2. If you have defined one dimension by percentage in your CSS, but not the other (for example, `width: 100%` needs `height: auto` to ensure the image grows in both dimensions). 
+   
+    3. **Oversized images:** If the image does not define a `srcset `and the intrinsic image is significantly larger than the rendered image, the directive will display a warning suggesting the use of the `srcset` and `sizes` attributes.
+    
+    4. **Image density:** The directive will throw an error if you try to include an image in the `srcset` with a pixel density of more than `3x`. Descriptors higher than `2x` are generally not recommended because it has the unintended consequence of forcing high-resolution mobile devices to download huge images. Moreover, the human eye can't actually tell much of a difference [above 2x](https://web.dev/uses-rel-preconnect/).
```

Comment:
> Moreover, the human eye can't actually tell much of a difference [above 2x](https://web.dev/uses-rel-preconnect/).

hmm that link doesn't seem right.

Reply:
There was some discussion about this. Have changed it to another link thats more relevant.